ethereum/bridge/contracts/Bridge.sol

```diff
@@ -0,0 +1,295 @@
+		bytes32 hash = ethSignedMessageHash(bridgeMessage);
+		for (uint256 i = 0; i < signatures.length; i++) {
+			address recoveredPK = recoverSigner(hash, signatures[i]);
+			// Check if the address is not zero
+			require(recoveredPK != address(0), 'Zero address.');
+			uint256 index = validatorIndex[recoveredPK] - 1;
+			require(index < validators.length, 'Index out of bounds');
+
+			Validator memory validator = validators[index];
+			require(recoveredPK == validator.addr, 'Invalid signature');
+			totalWeight += validator.weight;
+		}
+
+		// TODO
+		require(totalWeight >= 999, 'Not enough total signature weight');
+		if (bridgeMessage.messageType == 1) resumeBridge();
+	}
+
+	// Check also weight. i.e. no more than 33% of the total weight
+	// A function to add a validator
+	function addValidator(address _pk, uint256 _weight) private {
+		// Check if the address is not zero
+		require(_pk != address(0), 'Zero address.');
+		// Check if the address is not already a validator
+		require(validatorIndex[_pk] == 0, 'Already a validator.');
+		// Add the validator to the array
+		validators.push(Validator(_pk, _weight));
+		// Update the validator index
+		validatorIndex[_pk] = validators.length;
+	}
+
+	function validatorsCount() public view returns (uint count) {
+		return validators.length;
+	}
+
+	// The contract can be upgraded by the owner
+	function _authorizeUpgrade(address newImplementation) internal override {}
+
+	// Function to initiate a transfer from the source chain to the destination chain
+	function initiateTransfer(address recipient, uint256 amount) external {
+		// Transfer the tokens from the sender to this contract
+		// require(
+		//     IERC20(token).transferFrom(msg.sender, address(this), amount),
+		//     "Transfer failed"
+		// );
+	}
+
+	// Function to complete a transfer from the destination chain to the source chain
+	function completeTransfer(
+		address sender,
+		address recipient,
+		uint256 amount,
+		uint256 nonce,
+		bytes memory signature
+	) private {
+		// Verify that the nonce is correct
+		require(processedNonces[recipient][nonce] == false, 'transfer already processed');
+
+		// Increment the nonce for the recipient
+		processedNonces[recipient][nonce] = true;
+
+		// Emit the transfer completed event
+		emit TransferCompleted(sender, recipient, amount, nonce);
+	}
+
+	// returning the contract's balance in wei
+	function getBalance() public view returns (uint256) {
+		return address(this).balance;
+	}
+
+	function transfer(address payable transferAddress, uint256 amount) public {
+		transferAddress.transfer(amount);
+	}
+
+	// "0x93f82d7903c6a37336c33d68a890b448665735b4f513003cb4ef0029da0372b9329e0f6fc0b9f9c0c77d66bbf7217da260803fcae345a72f7a7764c56f464b5c1b"
+	// [1 ,2 ,3 ,4 ,"0x5567f54B29B973343d632f7BFCe9507343D41FCa" ,5 ,"0x5567f54B29B973343d632f7BFCe9507343D41FCa"]
+	function ethSignedMessageHash(
+		BridgeMessage calldata bridgeMessage
+	) public pure returns (bytes32) {
+		bytes32 hash = keccak256(
```

Comment:
This does not look right, we basically hash message twice: one in this line and one in `toEthSignedMessageHash`

Also note `toEthSignedMessageHash` (https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/utils/cryptography/MessageHashUtils.sol#L33) prefixes message with `\x19Ethereum Signed Message:\n32` which is not what we want to use. I know we talked about using ERC 191 standard earlier but we don't need this any more based on our discussion